app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterScreen.kt

```diff
@@ -0,0 +1,194 @@
+ * account. It navigates through multiple registration steps, each handling different parts of the
+ * registration process, including entering username, email, password, personal details, and
+ * uploading a profile picture.
+ *
+ * @param client ChatClient instance used for handling user connection.
+ * @param nav Navigation actions for handling back navigation and successful registration
+ *   completion.
+ * @param authViewModel ViewModel to manage and observe authentication related data.
+ * @param userViewModel ViewModel to manage user related data and operations.
+ * @param onNavToExplore Function to execute once registration completes successfully and user
+ *   navigates to Explore Screen.
+ */
+@Composable
+fun RegisterScreen(
+    client: ChatClient,
+    nav: NavigationActions,
+    authViewModel: AuthViewModel,
+    userViewModel: UserViewModel,
+    onNavToExplore: () -> Unit
+) {
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+  val signInState = authViewModel.signInState.collectAsState()
+  val textFieldColors =
+      TextFieldDefaults.colors(
+          focusedTextColor = Color.Black,
+          unfocusedTextColor = Color.Black,
+          unfocusedContainerColor = Color.Transparent,
+          focusedContainerColor = Color.Transparent,
+          cursorColor = Color.Black,
+          focusedLabelColor = MaterialTheme.colorScheme.tertiary,
+          focusedIndicatorColor = MaterialTheme.colorScheme.tertiary)
+
+  var state by remember { mutableIntStateOf(1) }
+
+  Column(modifier = Modifier.fillMaxSize()) {
+    TopAppBar(
+        modifier = Modifier.testTag("TopBar"),
+        backgroundColor = MaterialTheme.colorScheme.background,
+        elevation = 0.dp,
+        title = {},
+        navigationIcon = {
+          IconButton(
+              onClick = {
+                if (state == 0) {
+                  nav.goBack()
+                } else {
+                  state -= 1
+                }
+              }) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Back",
+                    tint = MaterialTheme.colorScheme.onBackground)
+              }
+        })
+
+    Column(
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier.fillMaxSize().padding(25.dp).verticalScroll(ScrollState(0))) {
+          Spacer(modifier = Modifier.height(screenHeight / 10))
+          Image(
+              painter = painterResource(id = R.drawable.gomeet_text),
+              contentDescription = "Go Meet",
+              modifier = Modifier.padding(top = 0.dp),
+              alignment = Alignment.Center,
+              colorFilter = ColorFilter.tint(MaterialTheme.colorScheme.tertiary))
+
+          Spacer(modifier = Modifier.height(screenHeight / 10))
+
+          fun nextState() {
```

Comment:
This is a good idea, but upon going back a screen, the fields start out as empty again

Reply:
Fixed !